atlasopenmagic/metadata.py

```diff
@@ -1,364 +1,384 @@
-    'dataset_id': 'dataset_number',
-    'short_name': 'physics_short',
-    'e-tag': 'e-tag',
-    'cross_section': 'crossSection_pb',
-    'filter_efficiency': 'genFiltEff',
-    'k_factor': 'kFactor',
-    'number_events': 'nEvents',
-    'sum_weights': 'sumOfWeights',
-    'sum_weights_squared': 'sumOfWeightsSquared',
-    'process': 'process',
-    'generators': 'generator',
-    'keywords': 'keywords',
-    'description': 'description',
-    'job_link': 'job_path',
-}
+def _apply_protocol(url, protocol):
+    """
+    Internal helper to transform a root URL into the specified protocol format.
 
-# Set the metadata URL based on the current release
-_METADATA_URL = LIBRARY_RELEASES[current_release]
+    Args:
+        url (str): The base 'root://' URL.
+        protocol (str): The target protocol ('https', 'eos', or 'root').
+
+    Returns:
+        str: The transformed URL.
+    """
+    if protocol == 'https':
+        # Convert to a web-accessible URL via opendata.cern.ch
+        return url.replace(
+            'root://eospublic.cern.ch:1094/',
+            'https://opendata.cern.ch')
+    if protocol == 'eos':
+        # Provide the path relative to the EOS mount point
+        return url.replace('root://eospublic.cern.ch:1094/', '')
+    if protocol == 'root':
+        # Return the original URL for direct ROOT access
+        return url
+    raise ValueError(
+        f"Invalid protocol '{protocol}'. Must be 'root', 'https', or 'eos'.")
+
+
+def _fetch_and_cache_release_data(release_name):
+    """
+    Internal helper to fetch all datasets for a release and populate the local cache.
+    This function performs a single, efficient API call to `/releases/{release_name}`
+    to minimize network latency.
+
+    Args:
+        release_name (str): The name of the release to fetch.
+
+    Raises:
+        ValueError: If the API call fails or returns an error.
+    """
+    global _metadata
+    print(f"Fetching and caching all metadata for release: {release_name}...")
+    try:
+        # Call the API endpoint that returns the full release details,
+        # including all datasets.
+        response = requests.get(f"{API_BASE_URL}/releases/{release_name}", timeout=300)
+        response.raise_for_status()  # Raise an exception for bad status codes (4xx or 5xx)
+        release_data = response.json()
+
+        # Create a new cache dictionary. This allows for an atomic update of
+        # the global cache.
+        new_cache = {}
+        # Iterate through the datasets returned by the API.
+        for dataset in release_data.get('datasets', []):
+            # Cache the dataset by its unique number (as a string).
+            ds_number_str = str(dataset['dataset_number'])
+            new_cache[ds_number_str] = dataset
+            # Also cache by the physics short name, if available, for user
+            # convenience.
+            if dataset.get('physics_short'):
+                new_cache[dataset['physics_short']] = dataset
```

Comment:
```suggestion
                new_cache[dataset['physics_short'].lower()] = dataset
```

Thinking about how to keep all of this case-insensitive. I don't remember most days if it's MGPyEG or MGPyEg... SH or Sh?